src/ansible_navigator/actions/open_file.py

```diff
@@ -98,26 +98,32 @@ def run(self, interaction: Interaction, app: AppPublic) -> None:
                 return None
 
         if not filename:
+            # common arguments, only suffix varies
+            myargs = {'mode': 'w', 'encoding': 'utf-8', 'delete': False}
             if interaction.ui.serialization_format() == "text.html.markdown":
-                with tempfile.NamedTemporaryFile(suffix=".md", delete=False) as temp_file:
+                myargs['suffix'] = '.md'
+                with tempfile.NamedTemporaryFile(**myargs) as temp_file:
+                    temp_file.write(obj)
                     filename = temp_file.name
-                    with open(filename, "w", encoding="utf-8") as fh:
-                        fh.write(obj)
             elif interaction.ui.serialization_format() == "source.yaml":
-                with tempfile.NamedTemporaryFile(suffix=".yaml", delete=False) as temp_file:
+                myargs['suffix'] = '.yaml'
+                with tempfile.NamedTemporaryFile(**myargs) as temp_file:
+                    human_dump(obj=obj, filename=temp_file)
                     filename = temp_file.name
-                    human_dump(obj=obj, filename=filename)
             elif interaction.ui.serialization_format() == "source.json":
-                with tempfile.NamedTemporaryFile(suffix=".json", delete=False) as temp_file:
+                myargs['suffix'] = '.json'
```

Comment:
Why `delete=False` is removed? This will delete the temp file outside the scope. The temp file is needed later on when it is opened on the editor